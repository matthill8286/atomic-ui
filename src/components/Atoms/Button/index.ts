export * from './Button'
export * from './Button.interface'
export * from './IconButton'
