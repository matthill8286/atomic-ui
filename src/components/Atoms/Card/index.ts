export * from './CardV2'
export * from './Card.interface'
export { StyledCard } from './Card.styled'
