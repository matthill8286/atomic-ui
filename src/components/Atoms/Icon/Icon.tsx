import { FlexAlignItemsOptions } from '@/components/Helper/FlexBox'
import { padding } from '@/styles/sc-shared-functions'
import { css, styled } from '@/styles/styled'
import { PaddingProps, Size, ThemeColors } from '@/types'

export type RotationValues = 0 | 45 | 90 | 180 | 270 | -45 | -90 | -180 | -270

export interface IconProps {
  rotate?: RotationValues
  color?: ThemeColors
  withBorder?: boolean
  padding?: PaddingProps['padding']

  width?: number | keyof Omit<Size, 'xxxs'>
  height?: number | keyof Omit<Size, 'xxxs'>
  animate?: boolean
  display?: string
  alignSelf?: FlexAlignItemsOptions
}

const DefaultSize = '25px'

export const Icon = styled.div<IconProps>(
  ({
    theme,
    color,
    width,
    height,
    rotate = 0,
    animate = false,
    display = 'flex',
    alignSelf,
    withBorder,
  }) => css`
    ${display && `display: ${display};`};
    ${padding};
    ${alignSelf && `align-self: ${alignSelf};`};
    ${withBorder &&
      `border: 2px solid ${theme.color.grey2};
        border-radius: 50%;
        padding: ${theme.spacing.base.xs};`}
    svg {
      width: ${width ? getSizeInPx(width, theme.spacing.base) : DefaultSize};
      height: ${height ? getSizeInPx(height, theme.spacing.base) : DefaultSize};
      transform: rotate(${rotate}deg);
      fill: ${color ? theme.color[color] : 'currentColor'};
      ${animate &&
        css`
          transition: transform 0.3s ease-in-out;
        `};
    }
    svg g,
    svg p {
      fill: ${color ? theme.color[color] : 'currentColor'};
    }
  `
)

const getSizeInPx = (
  size: number | keyof Omit<Size, 'xxxs'>,
  themeSizes: Omit<Size, 'xxxs'>
): string => {
  if (typeof size === 'number') {
    return `${size}px`
  } else {
    return themeSizes[size] || `${DefaultSize}px`
  }
}
