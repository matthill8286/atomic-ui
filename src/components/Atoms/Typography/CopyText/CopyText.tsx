import React from 'react'
import { BoxDimensions, ThemeFontSizes } from '@/types/theme'
import { Typo } from '../Typo'
import { FontSizeMap } from '../Typo/Typo.interface'
import { CopyTextProps } from './CopyText.interface'

const getMargins = (fontSize?: ThemeFontSizes): BoxDimensions => {
  if (fontSize && (fontSize === 'xs' || fontSize === 'sm')) {
    return 'xs 0 md'
  } else {
    return ''
  }
}

export const CopyText: React.FC<CopyTextProps> = ({
  tag = 'p',
  withMargins = false,
  bold = false,
  color = 'grey5',
  fontSize = 'xs',
  children,
  margin,
  ...restProps
}) => (
  <Typo
    tag={tag}
    color={color}
    fontSize={fontSize}
    weight={bold ? 'bold' : 'regular'}
    margin={
      (withMargins &&
        fontSize &&
        (typeof fontSize === 'string'
          ? getMargins(fontSize)
          : getMargins((fontSize as FontSizeMap)?.mobile))) ||
      margin
    }
    {...restProps}>
    {children}
  </Typo>
)

CopyText.displayName = 'CopyText'
