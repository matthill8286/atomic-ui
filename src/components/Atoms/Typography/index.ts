export * from './Typo'
export * from './CopyText'
export * from './InfoText'
export * from './Heading'
