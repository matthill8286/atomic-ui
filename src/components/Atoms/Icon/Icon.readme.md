## How to use?
Import the `Icon` component and also import the SVG you want to use directly from the `@excelwithbusiness/webmobile-svg-library` module and nest it. The `Icon` component itself has the Prop `rotate` to allow a rotation of a SVG.

```javascript
import { Icon } from '@/components/Atoms/Icon'
import { IconArrow } from '@excelwithbusiness/webmobile-svg-library'

<Icon>
  <IconArrow />
</Icon>

<Icon rotate={90}> 
  <IconArrow />
</Icon>
```

  &nbsp;

## List of existing Icons

```js

IconAdd

IconArticle

IconBackUi

IconBookmark

IconBookmarkActive

IconCheckmarkCircle

IconCheckmarkCircleOutlined

IconCheckmarkSquare

IconCheckmarkSquareOutlined

IconClear

IconClearCircle

IconClearCircleOutlined

IconClearSquare

IconClearSquareOutlined

IconCross

IconDone

IconFastForward

IconFastRewind

IconFilter

IconForward

IconHeart

IconHeartOutlined

IconHelp

IconHelpOutlined

IconInfo

IconInfoOutined

IconMenuIcon

IconNext

IconPause

IconPlay

IconPlayOutlined

IconSearch

IconSettings

IconSettingsOutlined

IconSkip

IconSkipNext

IconSkipPrevious

IconVolumeDown

IconVolumeLow

IconVolumeMinus

IconVolumeMute

IconVolumeOff

IconVolumePlus

IconVolumeUp

```
