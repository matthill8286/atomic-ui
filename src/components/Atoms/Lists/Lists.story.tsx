import { Meta, Story } from '@storybook/react/types-6-0'
import * as React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography/CopyText'
import { Lists, ListsProps } from './Lists'
import { colorsList } from '@/utils/helper'
import { StyleguideCheckmarkCircle } from '@matthill8286/atomic-icon-library'

const ListItems = [
  'Lorem ipsum Lorem ipsum Lorem ipsum Lorem ipsum Lorem ',
  'Dolore est',
  'Testing this',
]

// eslint-disable-next-line import/no-default-export
export default {
  title: 'Design System/Atoms/Lists',
  component: Lists,
  argTypes: {
    color: {
      control: {
        type: 'select',
        options: colorsList,
      },
    },
  },
} as Meta

const Template: Story<ListsProps> = (args: ListsProps) => (
  <Lists {...args}>
    {ListItems.map((item, index) => (
      <li key={index}>
        {args.icon && (
          <Icon color="success" width={24} height={24}>
            <StyleguideCheckmarkCircle />
          </Icon>
        )}
        {<CopyText>{item}</CopyText>}
      </li>
    ))}
  </Lists>
)

export const Default = Template.bind({})
