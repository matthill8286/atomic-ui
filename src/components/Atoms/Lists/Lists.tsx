import React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { media } from '@/styles/media'
import { styled } from '@/styles/styled'
import { ThemeColors } from '@/types'

interface StyledListsProps {
  icon?: boolean
  ordered?: boolean
  withMargin?: boolean
  color?: ThemeColors
}

export interface ListsProps extends React.HTMLProps<HTMLElement> {
  icon?: boolean
  ordered?: boolean
  children: React.ReactNode
  withMargin?: boolean
  color?: ThemeColors
}

const StyledList = styled.ul<StyledListsProps>`
  margin: ${({ theme, withMargin }) =>
    withMargin ? `${theme.spacing.base.xs} 0 ${theme.spacing.base.md}` : '0'};
  padding: ${({ theme, icon }) => (icon ? '0' : `0 0 0 ${theme.spacing.base.sm}`)};
  font-size: ${({ theme }) => theme.font.size.xs};

  ${({ icon }) => icon && 'list-style-type: none;'}
  ${({ theme, color }) => color && `color: ${theme.color[color]};`}

  ${media.md} {
    font-size: ${({ theme }) => theme.font.size.sm};

    ${Icon} {
      top: 0;
    }
  }

  & li {
    margin: 0;
    padding: 0;
    margin-bottom: ${({ theme }) => theme.spacing.base.xs};

    & p:first-child {
      margin-top: 0;
    }

    & p:last-child {
      margin-bottom: 0;
    }
  }

  ${Icon} {
    float: left;
    position: relative;
    top: -2px;
    margin-right: ${({ theme }) => theme.spacing.base.xs};
  }
`

export const Lists: React.FC<ListsProps> = (props) => {
  const { ordered = false, icon = false, withMargin = false, children, className, color } = props
  return (
    <StyledList
      as={ordered ? 'ol' : 'ul'}
      icon={icon}
      withMargin={withMargin}
      className={className}
      color={color}>
      {children}
    </StyledList>
  )
}
