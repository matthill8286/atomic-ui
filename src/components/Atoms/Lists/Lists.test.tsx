import React from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { CopyText } from '@/components/Atoms/Typography/CopyText'
import { IconCheckmarkCircle } from '@excelwithbusiness/webmobile-svg-library'
import { renderWithTheme } from '@/testRenderer'
import { Lists } from './Lists'

describe('Lists', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <Lists>
        <li>
          <CopyText>Lorem ipsum</CopyText>
        </li>
      </Lists>
    )
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with icon', () => {
    const tree = renderWithTheme(
      <Lists>
        <li>
          <CopyText>
            <Icon color="success" height={24} width={24}>
              <IconCheckmarkCircle />
              Lorem ipsum
            </Icon>
          </CopyText>
        </li>
      </Lists>
    )
    expect(tree).toMatchSnapshot()
  })
})
