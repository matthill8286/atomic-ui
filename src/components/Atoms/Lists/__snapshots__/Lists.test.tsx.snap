// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists renders correctly 1`] = `
.c1 {
  color: #272422;
  font-size: 14px;
  line-height: 20px;
  font-family: SourceSansPro,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c1 a,
.c1 a:focus,
.c1 a:hover,
.c1 a:active,
.c1 a:visited {
  color: inherit;
}

.c0 {
  margin: 0;
  padding: 0 0 0 16px;
  font-size: 14px;
}

.c0 li {
  margin: 0;
  padding: 0;
  margin-bottom: 8px;
}

.c0 li p:first-child {
  margin-top: 0;
}

.c0 li p:last-child {
  margin-bottom: 0;
}

.c0 .Icon-sc-1vrq823-0 {
  float: left;
  position: relative;
  top: -2px;
  margin-right: 8px;
}

@media (min-width:752px) {
  .c0 {
    font-size: 16px;
  }

  .c0 .Icon-sc-1vrq823-0 {
    top: 0;
  }
}

<ul
  className="c0"
>
  <li>
    <p
      className="c1"
      color="grey5"
      fontSize="xs"
    >
      Lorem ipsum
    </p>
  </li>
</ul>
`;

exports[`Lists renders correctly with icon 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 svg {
  cursor: inherit;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #70c118;
}

.c1 {
  color: #272422;
  font-size: 14px;
  line-height: 20px;
  font-family: SourceSansPro,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c1 a,
.c1 a:focus,
.c1 a:hover,
.c1 a:active,
.c1 a:visited {
  color: inherit;
}

.c0 {
  margin: 0;
  padding: 0 0 0 16px;
  font-size: 14px;
}

.c0 li {
  margin: 0;
  padding: 0;
  margin-bottom: 8px;
}

.c0 li p:first-child {
  margin-top: 0;
}

.c0 li p:last-child {
  margin-bottom: 0;
}

.c0 .c2 {
  float: left;
  position: relative;
  top: -2px;
  margin-right: 8px;
}

@media (min-width:752px) {
  .c0 {
    font-size: 16px;
  }

  .c0 .c2 {
    top: 0;
  }
}

<ul
  className="c0"
>
  <li>
    <p
      className="c1"
      color="grey5"
      fontSize="xs"
    >
      <div
        className="c2 c3"
        color="success"
        height={24}
        width={24}
      >
        <svg
          height={32}
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="M12 2a10 10 0 1010 10A10 10 0 0012 2zm4.35 7.73l-5.62 5.62a.48.48 0 01-.7 0L7.65 13a.5.5 0 010-.71l.53-.53a.48.48 0 01.7 0l1.5 1.49 4.74-4.74a.5.5 0 01.7 0l.53.53a.5.5 0 010 .69z"
          />
        </svg>
        Lorem ipsum
      </div>
    </p>
  </li>
</ul>
`;
